quickwit/quickwit-actors/src/actor_handle.rs

```diff
@@ -208,13 +208,10 @@ impl<A: Actor> ActorHandle<A> {
 
     /// Waits until the actor exits by itself. This is the equivalent of `Thread::join`.
     pub async fn join(self) -> (ActorExitStatus, A::ObservableState) {
-        let exit_status = self.join_handle.await.unwrap_or_else(|join_err| {
-            if join_err.is_panic() {
-                ActorExitStatus::Panicked
-            } else {
-                ActorExitStatus::Killed
-            }
-        });
+        let exit_status =
+            self.join_handle.join().await.expect(
+                "Attempting to call join after universe.join() or universe.quit() was called",
```

Comment:
Two problems here:
- `.join()` can be called by someone else than `universe.join()` or `universe.quit()` so the error message is wrong.
- We don't want to bubble up panics, here. We want panics happening in actors to be intercepted and processed properly.
That's what the code was doing before.

Reply:
Not sure I understand the first point. We move self into join, so someone can only call join once. So, the only situation where someone calls the join and doesn't find join handle there and this error is thrown is if universe.join() was called before hand. 

Regarding the second point, if panic had occurred, we have already bubbled it up in universe.join().

Reply:
I think I was missing the fact that you swapped the tokio taask JoinHandle by your a new struct. Let me dig into that further.

Reply:
(you can hit the mark as resolved button, i am not doing to maximize the chance you read this :) )

Reply:
Yeah, it is a bit tricky structure because join handle consumes self and we need to share it between two places - actor handle and universe but can only be joined once.